pkg/antctl/raw/check/installation/command.go

```diff
@@ -0,0 +1,404 @@
+	if err != nil {
+		return fmt.Errorf("unable to create Kubernetes client: %s", err)
+	}
+	ctx := context.Background()
+	testContext := NewTestContext(client, config, clusterName, o)
+	if err := testContext.setup(ctx); err != nil {
+		return err
+	}
+	for name, test := range testsRegistry {
+		testContext.Header("Running test: %s\n", name)
+		if err := test.Run(ctx, testContext); err != nil {
+			testContext.Header("Test %s failed: %s", name, err)
+		} else {
+			testContext.Header("Test %s passed", name)
+		}
+	}
+	testContext.Log("Test finished")
+	testContext.teardown(ctx)
+	return nil
+}
+
+func agnhostConnectCommand(target string) []string {
+	return []string{"/agnhost", "connect", target, "--timeout=5s"}
+}
+
+func newService(name string, selector map[string]string, port int) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{Name: name, Port: int32(port)},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+type deploymentParameters struct {
+	Name        string
+	Role        string
+	Image       string
+	Replicas    int
+	Port        int
+	Command     []string
+	Affinity    *corev1.Affinity
+	Tolerations []corev1.Toleration
+	Labels      map[string]string
+}
+
+func newDeployment(p deploymentParameters) *appsv1.Deployment {
+	if p.Replicas == 0 {
+		p.Replicas = 1
+	}
+	replicas32 := int32(p.Replicas)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: p.Name,
+			Labels: map[string]string{
+				"name": p.Name,
+				"kind": p.Role,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: p.Name,
+					Labels: map[string]string{
+						"name": p.Name,
+						"kind": p.Role,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: p.Name,
+							Env: []corev1.EnvVar{
+								{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)},
+							},
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: int32(p.Port)},
+							},
+							Image:           p.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+						},
+					},
+					Affinity:    p.Affinity,
+					Tolerations: p.Tolerations,
+				},
+			},
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": p.Name,
+					"kind": p.Role,
+				},
+			},
+		},
+	}
+}
+
+func NewTestContext(client kubernetes.Interface, config *rest.Config, clusterName string, o *options) *testContext {
+	return &testContext{
+		client:          client,
+		config:          config,
+		clusterName:     clusterName,
+		antreaNamespace: o.antreaNamespace,
+		namespace:       generateRandomNamespace(testNamespacePrefix),
+	}
+}
+
+func generateRandomNamespace(baseName string) string {
```

Comment:
We may consider to add a new util function to generate random sequence and reuse it here.
I saw there are already two code pieces to generate random strings in Antrea. @antoninbas @tnqn any suggestion?

Reply:
this is yet to resolve

Reply:
This is a good idea, but let's handle this as a follow-up PR. There is indeed a version of this in `test/e2e` which I think is better.

Reply:
Sure